include/boost/gil/image_processing/numeric.hpp

```diff
@@ -38,7 +38,8 @@ inline double normalized_sinc(double x)
 /// otherwise: normalized_sinc(x) / normalized_sinc(x / a)
 inline double lanczos(double x, std::ptrdiff_t a)
 {
-    if (x == 0)
+    // means == but <= avoids compiler warning
+    if (0 <= x && x <= 0)
```

Comment:
Don't we have an `is_equal()` function for floating-point numbers ? We probably should...

Reply:
I am not aware of such function.

There are FP comparison tools in Boost.Test, https://www.boost.org/doc/libs/1_70_0/libs/math/doc/html/utils.html, but those may impose a performance hit. So, I went for a minimal compiler-friendly way.

If you or @simmplecoder prefer, feel free to add `is_equal`, then I'll refactor all FP tests from `==` to use such function.

Reply:
FYI, @simmplecoder, alternative implementations in GDAL do this

https://github.com/OSGeo/gdal/blob/b1c9c12ad373e40b955162b45d704070d4ebf7b0/gdal/alg/gdalwarpkernel.cpp#L3020-L3023

https://github.com/OSGeo/gdal/blob/b1c9c12ad373e40b955162b45d704070d4ebf7b0/gdal/alg/gdalwarpkernel_opencl.cpp#L1145-L1150